bindings/wasm/src/did/wasm_core_document.rs

```diff
@@ -452,12 +508,49 @@ impl WasmCoreDocument {
     let method_query: String = methodQuery.into_serde().wasm_result()?;
     let options: ProofOptions = options.0.clone();
 
-    let signer = self.0.signer(&private_key);
+    let guard = self.0.blocking_read();
+    let signer = guard.signer(&private_key);
     let signer = signer.options(options);
     let signer = signer.method(&method_query);
     signer.sign(&mut data).wasm_result()?;
     JsValue::from_serde(&data).wasm_result()
   }
+
+  // ===========================================================================
+  // Cloning
+  // ===========================================================================
+
+  /// Deep clones the `CoreDocument`.
+  #[wasm_bindgen(js_name = clone)]
+  pub fn deep_clone(&self) -> WasmCoreDocument {
+    WasmCoreDocument(Rc::new(CoreDocumentLock::new(self.0.blocking_read().clone())))
+  }
+
+  /// ### Warning
+  /// This is for internal use only. Do not rely on or call this method.
+  #[wasm_bindgen(js_name = _shallowCloneInternal)]
+  pub fn shallow_clone(&self) -> WasmCoreDocument {
+    WasmCoreDocument(self.0.clone())
+  }
+
+  // ===========================================================================
+  // Serialization
+  // ===========================================================================
+
+  /// Serializes this to a JSON object.
```

Comment:
technically it a JS object, but the `toJSON()` is a bingen IIRC

Reply:
Would it be better with "Serializes this to JSON"?  (or perhaps JSON string?)

Reply:
It does neither of those, that is the problem. JSON is the stringified format, but in this case it's plain JS objects.

Reply:
So we could keep the method name (to align with bindgen), but document it as `Serializes to plain JS representation` or similar.